File: src/main/java/kurenai/mybot/handler/impl/ForwardHandlerProperties.java
package kurenai.mybot.handler.impl;

import kurenai.mybot.handler.HandlerProperties;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

import javax.annotation.PostConstruct;
import java.util.HashMap;
import java.util.Map;

@Getter
@Setter
@ConfigurationProperties(prefix = "bot.handler.forward")
public class ForwardHandlerProperties extends HandlerProperties {

    private Group group;

    @PostConstruct
    public void init() {
        if (group.qqTelegram == null) group.qqTelegram = new HashMap<>();
        if (group.telegramQq == null) group.telegramQq = new HashMap<>();

        group.qqTelegram.forEach((k, v) -> group.telegramQq.putIfAbsent(v, k));
        group.telegramQq.forEach((k, v) -> group.qqTelegram.putIfAbsent(v, k));
    }

    @Getter
    @Setter
    static class Group {
        private Long            defaultQQ;
        private Long            defaultTelegram;
        private Map<Long, Long> qqTelegram;
        private Map<Long, Long> telegramQq;
    }
}

File: src/main/java/kurenai/mybot/handler/config/ForwardHandlerProperties.kt
package kurenai.mybot.handler.config

import kurenai.mybot.handler.HandlerProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import java.util.*

@ConstructorBinding
@ConfigurationProperties(prefix = "bot.handler.forward")
data class ForwardHandlerProperties(var master: Long = 0, var group: Group, var member: Member) : HandlerProperties() {

    data class Group(
        var defaultQQ: Long = 0, var defaultTelegram: Long = 0,
        var qqTelegram: MutableMap<Long, Long> = Collections.emptyMap(),
        var telegramQq: MutableMap<Long, Long> = Collections.emptyMap(),
    ) {
        init {
            qqTelegram.forEach { (k: Long, v: Long) -> telegramQq.putIfAbsent(v, k) }
            telegramQq.forEach { (k: Long, v: Long) -> qqTelegram.putIfAbsent(v, k) }
        }
    }

    data class Member(var bindingName: MutableMap<Long, String> = Collections.emptyMap())
}